src/Illuminate/Container/Container.php

```diff
@@ -234,6 +236,10 @@ public function bind($abstract, $concrete = null, $shared = false)
         // bound into this container to the abstract type and we will just wrap it
         // up inside its own Closure to give us more convenience when extending.
         if (! $concrete instanceof Closure) {
+            if (! is_string($concrete)) {
+                throw new \InvalidArgumentException('Concrete implementation must be a string or closure');
```

Comment:
If you throw a type error here, you don't need to add any throws doc, since `@throws Error` is already implicit for all methods.

```
throw new \TypeError(self::class.'::bind(): Argument #2 ($concrete) must be of type Closure|string|null');
```

Reply:
Sure, that seems like a good improvement 👍